### PR TITLE
[Merged by Bors] - feat: `0 < n ^ n` for all `n : ℕ`

### PR DESCRIPTION
**Pull request by YaelDillies**
From GrowthInGroups (LeanCamCombi)

Co-authored-by: Andrew Yang

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Nat/Defs.lean [hunk 1]
```diff
@@ -670,6 +670,15 @@ protected lemma div_pow (h : a ∣ b) : (b / a) ^ c = b ^ c / a ^ c := by
   refine (Nat.div_eq_of_eq_mul_right (pos_pow_of_pos c ha) ?_).symm
   rw [← Nat.mul_pow, Nat.mul_div_cancel_left' h]
 
+lemma pow_self_pos : ∀ n : ℕ, 0 < n ^ n
```
Comment (b-mehta) [inline Mathlib/Data/Nat/Defs.lean:673]: Can we also have a lemma describing exactly when m ^ n is positive (ie Nat.pow_pos_iff : 0 < m ^ n iff m > 0 or n = 0)? Then this one will be `simp [Nat.pow_pos_iff]`

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Data/Nat/Defs.lean [hunk 2]
```diff
@@ -670,6 +671,23 @@ protected lemma div_pow (h : a ∣ b) : (b / a) ^ c = b ^ c / a ^ c := by
...
 
+protected lemma pow_pos_iff : 0 < a ^ n ↔ 0 < a ∨ n = 0 where
+  mp h := by
+    rw [Nat.pos_iff_ne_zero] at h ⊢
+    contrapose! h
+    simp [h.1, h.2]
+  mpr := by
+    rintro (ha | rfl)
+    · exact Nat.pow_pos ha
+    · simp
```
Comment (tb65536) [inline Mathlib/Data/Nat/Defs.lean:682]: Golf using `Nat.pow_eq_zero`:
```suggestion
protected lemma pow_pos_iff : 0 < a ^ n ↔ 0 < a ∨ n = 0 := by
  rw [Nat.pos_iff_ne_zero, Nat.pos_iff_ne_zero, not_iff_comm, not_or, not_not, Nat.pow_eq_zero]
```

Comment (tb65536) [reply Mathlib/Data/Nat/Defs.lean:682]: Or with simp:
```suggestion
protected lemma pow_pos_iff : 0 < a ^ n ↔ 0 < a ∨ n = 0 := by
  simp [Nat.pos_iff_ne_zero, imp_iff_not_or]
```

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**